docs/slicer_cli_plugins.rst

```diff
@@ -0,0 +1,255 @@
+    },
+    "NucleiDetection": {
+      "type": "python"
+    },
+    "ComputeNucleiFeatures": {
+      "type": "python"
+    },
+    "NucleiClassification": {
+      "type": "python"
+    },
+    "BackgroundIntensity": {
+      "type": "python"
+    },
+    "SeparateStainsMacenkoPCA": {
+      "type": "python"
+    },
+    "PositivePixelCount": {
+      "type": "python"
+    },
+    "SuperpixelSegmentation": {
+      "type": "python"
+    }
+  }
+
+
+For each algorithm, doing:
+
+.. code-block:: bash
+
+  docker run {image/name} {algorithm} --xml
+
+returns xml that describes the algorithm and its command line interface.  For example ``docker run --rm dsarchive/histomicstk:latest NucleiDetection --xml`` generates (truncated as an example):
+
+.. code-block:: xml
+
+  <?xml version="1.0" encoding="UTF-8"?>
+  <executable>
+    <title>Detects Nuclei</title>
+    <description>Detects nuclei in a whole-slide image</description>
+    <category>HistomicsTK</category>
+    <version>0.1.0</version>
+    <documentation-url>https://digitalslidearchive.github.io/HistomicsTK/</documentation-url>
+    <license>Apache 2.0</license>
+    <contributor>Deepak Roy Chittajallu (Kitware), Neal Siekierski (Kitware)</contributor>
+    <acknowledgements>This work is part of the HistomicsTK project.</acknowledgements>
+    <parameters>
+      <label>IO</label>
+      <description>Input/output parameters</description>
+      <image>
+        <name>inputImageFile</name>
+        <label>Input Image</label>
+        <description>Input image</description>
+        <channel>input</channel>
+        <index>0</index>
+      </image>
+      <region>
+        <name>analysis_roi</name>
+        <label>Analysis ROI</label>
+        <description>
+          Region of interest within which the analysis should be done. Must be a four element
+          vector in the format "left, top, width, height" in the space of the base layer.  Default
+          value of "-1, -1, -1, -1" indicates that the whole image should be processed.
+        </description>
+        <longflag>analysis_roi</longflag>
+        <default>-1,-1,-1,-1</default>
+      </region>
+      <string-enumeration>
+        <name>nuclei_annotation_format</name>
+        <label>Nuclei annotation format</label>
+        <description>Format of the output nuclei annotations</description>
+        <longflag>nuclei_annotation_format</longflag>
+        <element>bbox</element>
+        <element>boundary</element>
+        <default>boundary</default>
+      </string-enumeration>
+      <file fileExtensions=".anot" reference="inputImageFile">
+        <name>outputNucleiAnnotationFile</name>
+        <label>Output Nuclei Annotation File</label>
+        <description>Output nuclei annotation file (*.anot)</description>
+        <channel>output</channel>
+        <index>1</index>
+      </file>
+    </parameters>
+    <parameters advanced="true">
+      ...
+    </parameters>
+  </executable>
+
+Requirements for the XML File
```

Comment:
This a bit confusing to me in that this section is titled 'Requirements ...' but then discusses imports and parsing. Maybe the main Heading for this section could be something like 'Required XML File'. Then further down before the 'Overall Section' could be another heading 'Requirements...' or 'Required Sections...'